### ISSUE TITLE
x/crypto/ssh/agent: TestMalformedRequests/Short_body failures on darwin/arm64 since 2021-12-08

### ISSUE THREAD

**Issue by bcmills** [golang/go#50161]
```
--- FAIL: TestMalformedRequests (60.00s)
    --- FAIL: TestMalformedRequests/Short_body (60.00s)
        client_test.go:230: ServeAgent returned expected error: read tcp 127.0.0.1:60723->127.0.0.1:60727: read: connection reset by peer
2021/12/14 08:02:42 agent 23: agent: incorrect passphrase
FAIL
FAIL	golang.org/x/crypto/ssh/agent	60.279s
```

`greplogs --dashboard -md -l -e 'FAIL: TestMalformedRequests'`

[2021-12-09T19:36:57-4570a08-de690c2/darwin-arm64-12_0-toothrot](https://build.golang.org/log/c1cddf7e5197a8ca5bed339de9f915e08dff95a6)
[2021-12-08T04:14:00-5770296-a19e72c/darwin-arm64-11_0-toothrot](https://build.golang.org/log/8e6b973963f3acd624ae4268b21b992115e1e2a6)

**Comment by bcmills** [comment]
This is a release-blocker via #11811.

Given the timing, it could be related to the fix for #49932 (CC @rolandshoemaker @julieqiu @katiehockman). Note that at least one of the failures is on `release-branch.go1.17`, so this does not appear to be a regression in Go 1.18 itself.

**Comment by cagedmantis** [comment]
Checking in on this as a release-blocking issue. Are there any updates?
